### PR TITLE
params: Use `ast.literal_eval` for python params.

### LINKED ISSUE #6953: Negative numbers from a Python file are not recognized as parameters
# Bug Report

## Description

When trying to do ```dvc run``` or ```dvc repro``` with a stage whose parameters include a negative number in a Python file there is this error:
```
ERROR: Parameters 'my_int' are missing from 'my_params.py'.           
```
However, when using YAML files instead of Python, everything works as expected.

### Reproduce

1. ```git init```
2. ```dvc init```
3. dvc.yaml
```
stages:
  my_stage:
    cmd: cat a.txt > b.txt
    deps:
    - a.txt
    outs:
    - b.txt
    params:
    - my_params.py:
      - my_int
```
4. my_params.py
```
my_int = -1
```
5. ```echo dummyString > a.txt```
6. ```dvc repro```

### Expected

Ideally I would expect this output (which can be achieved in this example by setting ```my_int``` to non-negative value):
```
Running stage 'my_stage':                                             
> cat a.txt > b.txt
Updating lock file 'dvc.lock' 
```

### Environment information

**Output of `dvc doctor`:**

```console
$ dvc doctor
DVC version: 2.8.2 (pip)
---------------------------------
Platform: Python 3.8.10 on Linux-5.11.0-37-generic-x86_64-with-glibc2.29
Supports:
	webhdfs (fsspec = 2021.10.1),
	http (aiohttp = 3.7.4.post0, aiohttp-retry = 2.4.5),
	https (aiohttp = 3.7.4.post0, aiohttp-retry = 2.4.5),
	s3 (s3fs = 2021.8.1, boto3 = 1.17.106)
Cache types: reflink, hardlink, symlink
Cache directory: xfs on /dev/sdb
Caches: local
Remotes: None
Workspace directory: xfs on /dev/sdb
Repo: dvc, git
```

### PR DESCRIPTION
**Pull request by daavoo**
Add unit tests for `parse_py`

Closes  #6953

* [X] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

`dvc.org` P.R: https://github.com/iterative/dvc.org/pull/3014

### CODE REVIEW

**Comment by efiop** [comment]
@aandrusenko @skshetry (from https://github.com/iterative/dvc/pull/4456) would you mind taking a look, please?

**Comment by aandrusenko** [comment]
I have no comments

File: tests/unit/utils/serialize/test_python.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import pytest
```
Comment (karajan1001) [inline tests/unit/utils/serialize/test_python.py:1]: The code part LGTM.

`ast.literal_eval` are quite stable. I don't consider it to be worthy to do a detailed test on it. But here the tests involve collections and sequences. I just wondered that we don't test them in previous?

Comment (daavoo) [reply tests/unit/utils/serialize/test_python.py:1]: We test them here: 

https://github.com/iterative/dvc/blob/master/tests/unit/dependency/test_params.py#L114

I went for a separate unit test as that one involves `ParamsDependency` and `Stage` in the mix. 

I could remove this file and just add a test case for the `UNARY_OP` in the file linked above.

**Comment by karajan1001** [comment]
@Mergifyio rebase

File: tests/unit/utils/serialize/test_python.py (hunk 1 above)
Comment (skshetry) [reply tests/unit/utils/serialize/test_python.py:1]: >  I don't consider it to be worthy to do a detailed test on it.

It's not testing `ast.literal_eval`, it's testing that we are converting assignment statement into a key-value pair correctly.

**Review by skshetry — APPROVED** [review]

File: tests/unit/utils/serialize/test_python.py (hunk 1 above)
Comment (karajan1001) [reply tests/unit/utils/serialize/test_python.py:1]: > > I don't consider it to be worthy to do a detailed test on it.
> 
> It's not testing `ast.literal_eval`, it's testing that we are converting assignment statement into a key-value pair correctly.

Yes, I know that ` But here the tests involve collections and sequences. I just wondered that we don't test them in previous?`

**Pull request merged**